[CU-86eucqg87] Upgrade all systems

adelphi-liong opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added automated CI/CD workflows for pre-commit checks, Docker builds, releases, and merge gatekeeping.
  * Introduced scripts for Docker image building, pre-commit execution, release automation, and container testing.
  * Added configuration files for Dependabot, gitlint, Prettier, and project state management.

* **Dependency Updates**
  * Upgraded core dependencies to "@atomicloud/cyan-sdk" v2.0.0 across JavaScript, TypeScript, Python, and .NET components.
  * Updated base Docker images and Python/Node/.NET package versions for plugins, processors, and templates.

* **Chores**
  * Standardized code style and formatting rules.
  * Enhanced Nix configurations for package management and development shells.
  * Added a new test task to automate container testing.
  * Adjusted semantic-release configuration for commit scopes and message templates.

* **Documentation**
  * Added comprehensive project and development documentation.

* **Refactor**
  * Improved import statements and variable declarations for clarity and consistency.
  * Updated template placeholders to disable dynamic insertion for maintainability.

* **Bug Fixes**
  * Corrected namespace imports in .NET and template files.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

adelphi-liong committed:
feat: upgrade all systems

kirinnee commented:
Task linked: [CU-86eucqg87 Migrate templates to new system](https://app.clickup.com/t/86eucqg87)

adelphi-liong committed:
fix(nix): add missing shells

- added ci shell with main and lint dependencies
- added releaser shell with main, lint, and releaser dependencies

the addition of ci and releaser shells improves the development environment by
providing necessary configurations for continuous integration and release
processes.

adelphi-liong commented on .github/workflows/⚡reusable-docker.yaml:
```diff
@@ -0,0 +1,64 @@
+          DOCKER_USER: ${{ github.actor }}
+
+          # downwards API
+          GITHUB_REPO_REF: ${{ github.repository }}
+          GITHUB_SHA: ${{ github.sha }}
+          GITHUB_BRANCH: ${{ env.GITHUB_REF_SLUG }}
+
```
`AtomiCloud/actions.setup-docker@v1`  and `AtomiCloud/actions.setup-nix@v2` sets it up

adelphi-liong commented on .github/workflows/⚡reusable-release.yaml:
```diff
@@ -0,0 +1,30 @@
+      - nscloud-cache-tag-${{ inputs.atomi_platform }}-${{ inputs.atomi_service }}-releaser-nix-store-cache
+    steps:
+      # Setup
+      - uses: AtomiCloud/actions.setup-nix@v2
+        with:
+          auth-bot-app-id: ${{ vars.AUTH_BOT_APP_ID }}
+          auth-bot-secret-key: ${{ secrets.AUTH_BOT_SECRET_KEY }}
+      - uses: AtomiCloud/actions.cache-npm@v1
+      - name: Release
```
pin to major is good enough for AtoomiCloud/actions as it is internally managed

adelphi-liong commented on .github/workflows/cd.yaml:
```diff
@@ -0,0 +1,41 @@
+    tags:
+      - 'v*.*.*'
+jobs:
+  #
+  docker:
+    name: Build Docker Image - ${{ matrix.image_name }}
+    strategy:
+      matrix:
```
it won't be identical -- it will be split by version name

adelphi-liong commented on nix/packages.nix:
```diff
@@ -1,48 +1,53 @@
-          sd
-          git
-          # lint
+          biome
+
           treefmt
-          gitlint
           shellcheck
 
-          # dotnet
-          # go
+          gitlint
+
           golangci-lint
           go
-
-
-          #infra
-          docker
           ;
 
         python = python312;
-        poetry = (poetry.override { python3 = python311; });
+        poetry = (poetry.override { python3 = python312; });
 
-        nodejs = nodejs_20;
-        dotnet = dotnet-sdk_9;
+        nodejs = nodejs_22;
+        dotnet = dotnet-sdk_8;
       }
     );
```
infisical exists on nixpkgs 25.05

adelphi-liong replied:
it exists: https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=infisical

adelphi-liong commented on plugin/python/requirements.txt:
```diff
@@ -1,15 +1,15 @@
-aiohttp==3.8.6
+aiohttp==3.9.5
```
too much breaking -- this can be handled different PR

adelphi-liong commented on scripts/ci/docker.sh:
```diff
@@ -0,0 +1,79 @@
+echo "📝 Generating Image tags..."
+
+# Obtain image
+IMAGE_ID="${DOMAIN}/${GITHUB_REPO_REF}/${CI_DOCKER_IMAGE//[._-]*$//}"
+IMAGE_ID=$(echo "${IMAGE_ID}" | tr '[:upper:]' '[:lower:]') # convert to lower case
+
+# obtaining the version
+SHA="$(echo "${GITHUB_SHA}" | head -c 6)"
+BRANCH="${GITHUB_BRANCH//[._-]*$//}"
+IMAGE_VERSION="${SHA}-${BRANCH}"
```
that's the intention

adelphi-liong commented on scripts/ci/docker.sh:
```diff
@@ -0,0 +1,79 @@
+echo "  ✅ Latest Image Ref: ${LATEST_IMAGE_REF}"
+[ "${version}" != '' ] && echo "  ✅ Semver Image Ref: ${SEMVER_IMAGE_REF}"
+
+echo "🔨 Building Dockerfile..."
+args=""
+if [ "$BRANCH" = "$LATEST_BRANCH" ]; then
+  echo "🔎 Detected branch is '${LATEST_BRANCH}', push 'latest' tag!"
+  args="-t ${LATEST_IMAGE_REF}"
+fi
+if [ "${version}" != '' ]; then
+  args="${args} -t ${SEMVER_IMAGE_REF}"
+fi
+
+# shellcheck disable=SC2086
+docker buildx build \
+  "${CI_DOCKER_CONTEXT}" \
+  -f "${CI_DOCKERFILE}" \
+  --platform=${CI_DOCKER_PLATFORM} \
+  --push \
+  -t "${COMMIT_IMAGE_REF}" $args \
+  -t "${BRANCH_IMAGE_REF}"
+echo "✅ Pushed branch image!"
```
this looks correct

adelphi-liong commented on scripts/ci/release.sh:
```diff
@@ -0,0 +1,4 @@
+#! /bin/sh
+
+rm .git/hooks/*
+sg release -i npm || true
```
git hooks should be removed

adelphi-liong replied:
since its in CI, git will alllways exists

adelphi-liong committed:
fix: PR comments

adelphi-liong commented:
@coderabbitai re-review and resolve conversations you think are closed
